tools/GenerateIntegrationDefinitions/Program.cs

```diff
@@ -0,0 +1,121 @@
+                                                             },
+                                                             wrapper = new
+                                                             {
+                                                                 assembly = integrationsAssembly.FullName,
+                                                                 type = item.wrapperType.FullName,
+                                                                 method = item.wrapperMethod.Name,
+                                                                 signature = GetMethodSignature(item.wrapperMethod)
+                                                             }
+                                                         }
+                               };
+
+            var serializerSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore,
+                Formatting = Formatting.Indented
+            };
+
+            var json = JsonConvert.SerializeObject(integrations, serializerSettings);
+            Console.WriteLine(json);
+
+            string filename = "integrations.json";
+
+            if (args.Length > 0 && !string.IsNullOrWhiteSpace(args[0]))
+            {
+                filename = args[0];
+            }
+
+            var utf8NoBom = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+            File.WriteAllText(filename, json, utf8NoBom);
+        }
+
+        private static string GetIntegrationName(Type wrapperType)
+        {
+            const string integrations = "Integration";
+            var typeName = wrapperType.Name;
+
+            if (typeName.EndsWith(integrations, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return typeName.Substring(startIndex: 0, length: typeName.Length - integrations.Length);
+            }
+
+            return typeName;
+        }
+
+        private static string GetMethodSignature(MethodInfo method)
+        {
+            var returnType = method.ReturnType;
+            var parameters = method.GetParameters().Select(p => p.ParameterType).ToArray();
+
+            var signatureHelper = SignatureHelper.GetMethodSigHelper(method.CallingConvention, returnType);
+            signatureHelper.AddArguments(parameters, requiredCustomModifiers: null, optionalCustomModifiers: null);
+            var signatureBytes = signatureHelper.GetSignature();
+
+            if (method.IsGenericMethod)
+            {
+                // if method is generic, or IMAGE_CEE_CS_CALLCONV_GENERIC into the
+                // first byte (calling convention) and insert second byte with generic parameter count
+                var genericArguments = method.GetGenericArguments();
+
+                var newSignatureBytes = new byte[signatureBytes.Length + 1];
+                newSignatureBytes[0] = (byte)(signatureBytes[0] | 0x10);
```

Comment:
Nit: would be nice to name 0x10 `IMAGE_CEE_CS_CALLCONV_GENERIC` or something similar for code readability